### ISSUE TITLE
Accessibility: SectionSelectorWidget List, Contrast, and ARIA Issues

### ISSUE THREAD

**Issue by jaydonkrooss** [tl-its-umich-edu/canvas-course-manager-next#702]
Problematic component: `SectionSelectorWidget.tsx`

Issue summary:
- Section list invalid role=button: `<List>` + `<ListItemButton>`
- Merge sections inactive sections: insufficient contrast on disabled item
- Sort button: unnecessary "popup" ARIA attributes

Based on these reported problems found in [CCM Accessibility Review](https://docs.google.com/spreadsheets/d/1mYrbqwFLLXQoGRTY_XEGkfS3cu3BO_pUN_zQ_DFdhUY/edit?gid=1709548063#gid=1709548063):

- [Severity: Critical] U-M Users One Section
    - The list of sections (#tool_content_31 ul) has role=button elements, which are not allowed as part of a `<ul>` tag.
- [Severity: Critical] Merge Sections
    - Review before Merge list (#tool_content_31 ul) has role=button elements, which are not allowed as part of a ul tag.
- [Severity: Critical] Non-UM User
    - Select the Section list (#tool_content_31 ul) has role=button elements, which are not allowed as part of a ul tag.
- [Severity: Critical] Merge Sections
    - Select section list (#tool_content_31 ul) has role=button elements, which are not allowed as part of a ul tag.
- [Severity: Critical] Format 3rd Party Gradebook
    - Select one or more section list (#tool_content_31 ul) has role=button elements, which are not allowed as part of a ul tag.
- [Severity: Critical] Merge Sections
    - The inactive section names in the right side "Review before merge" list and some of the text on the Search by Course Name section have poor contrast (2.1:1; should be at least 4.5:1). The inactive Add and Undo buttons have a contrast of only 1.7:1.
- [Severity: Critical] Merge Sections
    - NVDA announced "Menu button submenu sort" if this button does not open a menu or a popup, aria-haspopup="true" and aria-controls="simple-menu" should be removed from the button. Select All button is Okay.

**Comment by zqian** [comment]
Assigning to Matt since I will be out for a couple of weeks. Thanks, Matt!
